specification/postgresqlhsc/resource-manager/Microsoft.DBforPostgreSQL/stable/2022-11-08/postgresqlhsc.json

```diff
@@ -0,0 +1,2987 @@
+          "description": "The properties of a server in a cluster.",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "ClusterServerListResult": {
+      "description": "A list of servers in a cluster.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "description": "The list of servers in a cluster.",
+          "items": {
+            "$ref": "#/definitions/ClusterServer"
+          }
+        }
+      }
+    },
+    "ClusterForUpdate": {
+      "description": "Represents a cluster for update.",
+      "type": "object",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/ClusterPropertiesForUpdate",
+          "x-ms-client-flatten": true,
+          "description": "Properties of the cluster."
+        },
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Application-specific metadata in the form of key-value pairs."
+        }
+      }
+    },
+    "ClusterPropertiesForUpdate": {
+      "description": "The properties used to update a cluster.",
+      "type": "object",
+      "properties": {
+        "administratorLoginPassword": {
+          "type": "string",
+          "x-ms-secret": true,
+          "format": "password",
+          "x-ms-mutability": [
+            "create",
+            "update"
+          ],
+          "description": "The password of the administrator login."
+        },
+        "postgresqlVersion": {
+          "type": "string",
+          "description": "The PostgreSQL version of the cluster."
+        },
+        "citusVersion": {
+          "type": "string",
+          "description": "The Citus version of the cluster."
+        },
+        "enableShardsOnCoordinator": {
+          "type": "boolean",
+          "description": "If shards on coordinator is enabled or not for the cluster."
+        },
+        "enableHa": {
+          "type": "boolean",
+          "description": "If high availability is enabled or not for the cluster."
+        },
+        "preferredPrimaryZone": {
+          "type": "string",
+          "description": "Preferred primary zone for the cluster"
+        },
+        "coordinatorServerEdition": {
+          "type": "string",
+          "description": "The edition of the coordinator (default: GeneralPurpose)."
+        },
+        "coordinatorStorageQuotaInMb": {
+          "type": "integer",
+          "format": "int64",
+          "description": "The storage of the coordinator in MB."
+        },
+        "coordinatorVCores": {
+          "type": "integer",
+          "format": "int64",
+          "description": "The vCores count of the coordinator (max: 64)."
+        },
+        "coordinatorEnablePublicIpAccess": {
+          "type": "boolean",
+          "description": "If public IP is requested or not for the coordinator."
+        },
```

Comment:
recommend updating all these booleans to enum as explained here : 

[https://eng.ms/docs/products/arm/rp_onboarding/process/api_review_best_practices](https://eng.ms/docs/products/arm/rp_onboarding/process/api_review_best_practices)

Replace boolean/switch properties with better enum
A Boolean will forever have two valid values (true or false). A string enum type is always preferred. Also, properties should always provide better values just than True and False. For example two switches "isTypeA" and "isTypeB" should be replaced with one enum "type": [A, B, DefaultType]. Even if you still believe [True, False] are the correct values for a property, you should use a string enum with values [True, False] instead of boolean. Enums are always a more flexible and future proof option because they allow additional values to be added in the future in a non-breaking way, e.g. [True, False, Unknown].

Reply:
Changed boolean types to enum